index.js

```diff
@@ -15,13 +15,18 @@ function terser(userOptions = {}) {
     })
   );
 
+  let numOfBundles = 0;
```

Comment:
Could you use plugin context instead of closure?

Reply:
Thanks for pointing out plugin context. I didn't know it exists.
However, trying it out (basically attaching `numOfBundles` to `this`), makes the test hang, because an async operation is not cleaned up properly (worker is not cleaned up properly).
Could you please point me in the right direction or implement it yourself?
The only docs I've found on the context api is this: https://rollupjs.org/guide/en#context
However, it does not mention use of custom data.

I am also just curious why using context api is better than closure?
I am thankful for any clarification. 😊

```suggestion
return {
    name: "terser",

    renderStart() {
      if (!this.worker) {
        this.worker = new Worker(require.resolve("./transform.js"), {
          numWorkers: userOptions.numWorkers
        });
        this.numOfBundles = 0;
      }
      this.numOfBundles++;
    },

    renderChunk(code) {
      return this.worker.transform(code, minifierOptions).catch(error => {
        const { message, line, col: column } = error;
        console.error(
          codeFrameColumns(code, { start: { line, column } }, { message })
        );
        throw error;
      });
    },

    generateBundle() {
      this.numOfBundles--;
      // we only want to end worker on the last bundle
      if (this.numOfBundles == 0) {
        this.worker.end();
      }
    },

    renderError() {
      this.numOfBundles--;
      // we only want to end worker on the last bundle
      if (this.numOfBundles == 0) {
        this.worker.end();
      }
    }
  };
```